manual/manual/tutorials/coreexamples.etex

```diff
@@ -289,6 +289,87 @@ let rec insert x btree =
                 else Node(y, left, insert x right);;
 \end{caml_example}
 
+
+\subsection{Record and variant disambiguation}
+( This subsection can be skipped on the first reading )
+
+Astute readers may have wondered what happens when two or more record
+fields or constructors share the same name,
+
+\begin{caml_example*}{toplevel}
+type first =  { x : int; y:int; z:int }
+type middle = { x:int; y:int }
+type last =   { x: int };;
+type first_variant = A | B | C
+type last_variant = A;;
+\end{caml_example*}
+
+The answer is that when confronted with multiple options, OCaml
```

Comment:
This might be a bit confusing, since the rest of the section explains why this sentence is incorrect.  Also, I believe we should encourage people to rely on type annotation rather than on the order of definition to disambiguate references to labels/constructors.

What about explaining first that the locally available type information is used to disambiguate, and that the last defined label/constructor with the given name (in the current scope) is used as a fallback when no such information is available?

Also, it might be useful to explain that for record expressions and patterns, all fields are considered in the fallback (i.e. the last definition of a record type with *all* the listed fields is used).

Reply:
You are right reversing the order and exposing the type-directed disambiguation first and the fallback as a last resort yields a much more positive narrative than the current version.

> Also, it might be useful to explain that for record expressions and patterns, all fields are considered in the fallback

I will also try to re-emphasize this part of the examples.